### ISSUE TITLE
Zebra requires lock on data dir

### ISSUE THREAD

**Issue by idky137** [zingolabs/infrastructure#19]
Zebrad requires a lock on the data dir used to load chain data. This means that tests that load a chain cache must be run individually.

One option could be to copy the chain cache data into a tempdir for each test being.

**Comment by ala-mode** [comment]
Acknowledged.

**Comment by ala-mode** [comment]
This is a duplicate of issue #7 - closing that issue.

**Comment by nachog00** [comment]
@idky137 
How do you replicate this issue?
Would runs of `zebrad` consuming tests fail if they don't get the chain cache JUST like it came out of the cache generation process?
Would they fail with some particular error message if that is the failing cause?
Would they not fail at all but the test makes no sense anymore?

**Comment by zancas** [comment]
This issue seems to be related to https://github.com/zingolabs/zaino/issues/114 since that issue requires Zebra regtest to load a cached chain, and this one requires tests to be runnable in parallel even when the tests in question cause a zebrads to access cached chains.

@nachog00 or @ala-mode would it make sense to list the tests that are affected by chain cache locks in this issue, or in a doc-comment in the relevant test mod?

**Comment by idky137** [comment]
We don't have any tests specifically for this bug, we could write some if its needed?

The best way currently is to run `cargo nextest run state_service` ( you may have to run `cargo nextest run --run-ignored ignored-only state_service` as these tests are ignored in https://github.com/zingolabs/zaino/pull/187 ). 

You should see a big error string for  most of the tests that contains the following message:
```
The application panicked (crashed).\nMessage:  Opening database \"/home/idky137/.cache/zebra/state/v26/regtest\" failed: Error { message: \"IO error: While lock file: /home/idky137/.cache/zebra/state/v26/regtest/LOCK: Resource temporarily unavailable\" }. Hint: Check if another zebrad process is running. Try changing the state cache_dir in the Zebra config.
```
if you run these tests (`state_service`) with no_capture you should not ever hit this error (although there may be other failures currently on dev).
